x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorFeatures.java

```diff
@@ -231,9 +205,156 @@ public ConnectorFeatures build() {
                 filteringAdvancedConfig,
                 filteringRules,
                 incrementalSyncEnabled,
-                syncRulesAdvancedEnabled,
-                syncRulesBasicEnabled
+                syncRulesFeatures
             );
         }
     }
+
+    public static class FeatureEnabled implements ToXContentObject, Writeable {
```

Comment:
This was to deal with some technical debt in `Connector` representation of features in status quo. 

Some features are represented in the `features` object as:
```
"my_feauture": true
```

Whereas some are represented as:
```
"my_feature": {"enabled": true}
```

`FeatureEnabled` is helper class that helps solely with serialising (`toXContent`) and deserialising (`parse`) logic for the nested example. 

I felt that hacking the parser of `ConnnectorFeatures` class would be a bit ugly. I will add a comment explaining why we need the `FeatureEnabled`  class.

I double-checked the logic and I will change `Boolean` to `boolean` because the `enabled` property should not be nullable.